### ISSUE TITLE
slow_vector_initialization should catch capacity call

### ISSUE THREAD

**Issue by nazar-pc** [rust-lang/rust-clippy#8800]
### Summary

Call to capacity is DRY approach to fill pre-allocated vector and should be treated the same way as explicit length

### Lint Name

slow_vector_initialization

### Reproducer

I tried this code:

```rust
let mut vec1 = Vec::with_capacity(len);
vec1.resize(vec1.capacity(), 0);
```

I expected to see this happen:
```
14 |     let mut vec1 = Vec::with_capacity(len);
   |                    ----------------------- help: consider replace allocation with: `vec![0; len]`
15 |     vec1.resize(vec1.capacity(), 0);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(clippy::slow_vector_initialization)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#slow_vector_initialization
```
Instead, this happened:
No warning

### Version

```text
rustc 1.60.0-nightly (c5c610aad 2022-02-14)
binary: rustc
commit-hash: c5c610aad0a012a9228ecb83cc19e77111a52140
commit-date: 2022-02-14
host: x86_64-unknown-linux-gnu
release: 1.60.0-nightly
LLVM version: 13.0.0
```
